shell/edit/provisioning.cattle.io.cluster/Basics.vue

```diff
@@ -0,0 +1,738 @@
+          value:       initial,
+          unsupported: true,
+          disabled:    true
+        });
+      }
+
+      return out;
+    },
+
+    haveArgInfo() {
+      return Boolean(this.selectedVersion?.serverArgs && this.selectedVersion?.agentArgs);
+    },
+
+    serverArgs() {
+      return this.selectedVersion?.serverArgs || {};
+    },
+
+    agentArgs() {
+      return this.selectedVersion?.agentArgs || {};
+    },
+
+    /**
+     * The addons (kube charts) applicable for the selected kube version
+     *
+     * { [chartName:string]: { repo: string, version: string } }
+     */
+    chartVersions() {
+      return this.selectedVersion?.charts || {};
+    },
+
+    showCisProfile() {
+      return (this.provider === 'custom' || this.isElementalCluster) && ( this.serverArgs?.profile || this.agentArgs?.profile );
+    },
+
+    enabledSystemServices: {
+      get() {
+        const out = difference(this.serverArgs.disable.options, this.serverConfig.disable || []);
+
+        return out;
+      },
+
+      set(neu) {
+        const out = difference(this.serverArgs.disable.options, neu);
+
+        this.$emit('enabledSystemServicesChanged', out);
+      },
+    },
+
+    showCloudConfigYaml() {
+      if ( !this.agentArgs['cloud-provider-name'] ) {
+        return false;
+      }
+
+      const name = this.agentConfig['cloud-provider-name'];
+
+      if ( !name ) {
+        return false;
+      }
+
+      switch ( name ) {
+      case 'none': return false;
+      case 'aws': return false;
+      case 'rancher-vsphere': return false;
+      case HARVESTER: return false;
+      default: return true;
+      }
+    },
+
+    showVsphereNote() {
+      if ( !this.agentArgs['cloud-provider-name'] ) {
+        return false;
+      }
+
+      const name = this.agentConfig['cloud-provider-name'];
+
+      return name === 'rancher-vsphere';
+    },
+
+    showCni() {
+      return !!this.serverArgs.cni;
+    },
+
+    showCloudProvider() {
+      return this.agentArgs['cloud-provider-name'];
+    },
+
+    showk8s21LegacyWarning() {
+      const isLegacyEnabled = this.features(LEGACY);
+
+      if (!isLegacyEnabled) {
+        return false;
+      }
+      const selectedVersion = semver.coerce(this.value.spec.kubernetesVersion);
+
+      return semver.satisfies(selectedVersion, '>=1.21.0');
+    },
+
+    ciliumIpv6: {
+      get() {
+        // eslint-disable-next-line no-unused-vars
+        const cni = this.serverConfig.cni; // force this property to recalculate if cni was changed away from cilium and chartValues['rke-cilium'] deleted
+
+        return this.userChartValues[this.chartVersionKey('rke2-cilium')]?.cilium?.ipv6?.enabled || false;
+      },
+      set(val) {
+        this.$emit('ciliumIpv6Changed', val);
+      }
+    },
+
+    isHarvesterExternalCredential() {
+      return this.credential?.harvestercredentialConfig?.clusterType === 'external';
+    },
+
+    unsupportedCloudProvider() {
```

Comment:
This is _not_ defined in rke2.vue, but rke2.vue references it (L1972), so I think it needs to be moved there and passed in as a prop here.